### PR TITLE
change: adjust response format for V3 feature

### LINKED ISSUE #7320: feat: new API design for the Admin API
### Description

One of the plans in the APISIX V3 roadmap was discussed in the mailing list: new API design for the Admin API

mail list link: https://lists.apache.org/thread/jg46m1lxn2zx3gz1r77b3v6cz6v2l4bn

Now I'm going to implement it, and this task is split into the following subtasks:

- [ ] updated some list URIs(SSL, proto )
- [ ] response body format
- [ ] supports paging to get a list of objects such as route
- [ ] supports filter when getting a list of objects such as route
- [ ] additional filter fields supported by the Route object

This issue is used to track the progress of this task, and also used to record things that require collaboration with other projects (dashboard, apisix-ingress-controller)

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7320

It is sister PR of #7366

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/core/config_etcd.lua [hunk 1]
```diff
@@ -212,14 +212,15 @@ local function load_full_data(self, dir_res, headers)
         self:upgrade_version(item.modifiedIndex)
 
     else
-        if not dir_res.nodes then
-            dir_res.nodes = {}
+        local values = (dir_res and dir_res.nodes) or dir_res
```
Comment (spacewander) [inline apisix/core/config_etcd.lua:215]: Is `or dir_res` useful?
If dir_res is nil, we don't need `or dir_res`.

**Review by spacewander — APPROVED** [review]

File: apisix/core/config_etcd.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/core/config_etcd.lua:215]: `or dir_res` is useful.

when we run the test case of: https://github.com/apache/apisix/blob/master/t/core/etcd-sync.t#L198-L216

here `dir_res` = `res.body.list`, changed here: https://github.com/apache/apisix/pull/7630/files#diff-0dd73f147a1cb8d7591792a953ffd7714e5d41ef8123aeb681046b1dcbfef668R306

the data struct like:

```json
"body": {
	"list": {
            ...element
    }
}
```

and the `values` = `dir_res.list`


in other cases,  `dir_res` = `res.body.node`, the data struct like:

```json
"body": {
	"node": {
          "nodes": {
              ...element
          }
    }
}
```

the `values` = `dir_res.node.nodes`

`(dir_res and dir_res.nodes) or dir_res` is meant to be compatible with both.

Comment (spacewander) [reply apisix/core/config_etcd.lua:215]: Got it. Would you mind adding a comment?

Comment (tzssangglass) [reply apisix/core/config_etcd.lua:217]: add

**Review by spacewander — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
I have checked changes on md fiels and /t files, LGTM

**Pull request merged**